### PR TITLE
create initial draft of the Future Node updates section

### PR DESCRIPTION
**Pull request by raproid**
**What has changed:** Node updating instructions.
**How:** Added a draft of updating instructions to the README.md in the repo root. 
**Why:** Need to inform Node users on how to update their Node instance, as we're implementing a version check mechanism between Insolar Platform along with Insolar MainNet and client using them, and as we're also implementing operational restrictions as well.

### CODE REVIEW

**Comment by raproid** [comment]
Merging is blocked because I didn't add a description and a key reviewer before creating this PR.

**Comment by raproid** [comment]
This is a draft, so no rush. Information on metrics for the Node users to check is coming later this week. 

I also hope the guys will change the metric name: they misspelled the verb in its name.

**Comment by raproid** [comment]
Why the Autotests-observer-public (Awesome CI) failed, tho, I have no idea. I literally just switched to the master, pulled the latest changes and branched out about 2 hours ago. В чят призывается @Tsovak.

**Comment by Tsovak** [comment]
@dmitry-shidlovsky could you pls help @raproid to solve the problem with tests? the main problem is `invalid memory address or nil pointer dereference [recovered`

**Comment by raproid** [comment]
Dmitry said this is not a critical and I can still merge if I need to. So, everything is cool. Tnx!

File: README.md [hunk 1]
```diff
@@ -169,9 +169,24 @@ make migrate-init
    
 ## Future Node updates
 
-Upon any upcoming Insolar MainNet updates, the current Node version may suspend synching. To resume the synching, you need to update your Node instance.
+Upon any upcoming Insolar MainNet updates, the current Node version may suspend synching. To resume synching, you need to update your Node instance.
```
Comment (sofiayem) [inline README.md:172]: Version does not suspend anything. Node does. Also, remove "you need", just tell them to update it.
Suggestions:
Upon an upcoming Insolar MainNet update, the Node of the current version may suspend synching. To resume synching, update your Node instance.

File: README.md [hunk 2]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
 
-Update instructions for future versions of the Node are on the way and will be released closer to the next significant update. 
+Each time your Node instance connects to the trusted Heavy Material Node, your Node instance sends its client type (`client_type`), communication protocol version (`heavy_version`) and smart contract version (`contract_version`) in the metadata.
```
Comment (sofiayem) [inline README.md:174]: Metadata of what? Just drop it. Also, Oxford comma.
Suggestion:
Each time your Node instance connects to the trusted Heavy Material Node (HMN), the instance sends its client type (`client_type`), communication protocol version (`heavy_version`), and smart contract version (`contract_version`) to the node.

File: README.md [hunk 3]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+
+If the protocol version and smart contract version are lower than those accepted by the Heavy Material Node, the Heavy Material Node sends back the `version of the observer is outdated. Please upgrade this client` message.
```
Comment (sofiayem) [inline README.md:176]: The "blablabalbalbalblalbal" message is hard to parse.
Suggestion:
If the protocol and/or smart contract versions are older than those accepted by the HMN, the HMN sends back the following message: `version of the observer is outdated. Please upgrade this client`.
Btw, capitalization in the message is wrong.

File: README.md [hunk 4]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+
+Upon this message, your Node instance logs the message and exits in 12s signifying it needs to be updated to continue working proreply.
```
Comment (sofiayem) [inline README.md:178]: "Upon ..." presumes an event. Exits what? "proreply"?
Suggestion:
Upon receiving this message, your Node instance logs it and stops in 12s. Attempts to reconnect result in a stop.

File: README.md [hunk 5]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+
+Your Node instance also changes the `is_deprecated_client` metrics value to `1`, which causes a `Your Node is out of date. Please update your Node` message on the Grafana board used for the Node monitoring.
```
Comment (sofiayem) [inline README.md:180]: ... which causes the following message on the Grafana board used for the Node monitoring: `Your Node is out of date. Please update your Node`.

File: README.md [hunk 6]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+
+To update your Node instance:
+
+1. Exit your Node instance if it didn't exit automatically.
```
Comment (sofiayem) [inline README.md:184]: Exit -> Stop

File: README.md [hunk 7]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+If the protocol version and smart contract version are lower than those accepted by the Heavy Material Node, the Heavy Material Node sends back the `version of the observer is outdated. Please upgrade this client` message.
+
+Upon this message, your Node instance logs the message and exits in 12s signifying it needs to be updated to continue working proreply.
+
+Your Node instance also changes the `is_deprecated_client` metrics value to `1`, which causes a `Your Node is out of date. Please update your Node` message on the Grafana board used for the Node monitoring.
+
+To update your Node instance:
+
+1. Exit your Node instance if it didn't exit automatically. 
+2. Clone and build the Node as described in the [**Build** section](https://github.com/insolar/observer/blob/master/README.md#build) of this document. Alternatively, you can pull to the existing cloned repo and then build the Node.
```
Comment (sofiayem) [inline README.md:185]: Drop "of this document.". Link is enough.

File: README.md [hunk 8]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+3. Deploy the Node and Node API as described in the [**Configure and deploy the Node**](https://github.com/insolar/observer/blob/master/README.md#configure-and-deploy-the-node) and [**Configure and deploy the Node API**](https://github.com/insolar/observer/blob/master/README.md#configure-and-deploy-the-node-api) sections respectively.
+Check for [new or updated configuration parameters](https://github.com/insolar/observer/wiki/Configuration-parameters) to make sure your current configurations are fully compatible with the updated Node binary files.
```
Comment (sofiayem) [inline README.md:187]: This should be step 3 of the numbered list.
And the "3. Deploy..." should be step "4. Deploy...".

File: README.md [hunk 9]
```diff
@@ -169,9 +169,24 @@ make migrate-init
...
+Your Node instance also changes the `is_deprecated_client` metrics value to `1`, which causes a `Your Node is out of date. Please update your Node` message on the Grafana board used for the Node monitoring.
+
+To update your Node instance:
+
+1. Exit your Node instance if it didn't exit automatically. 
+2. Clone and build the Node as described in the [**Build** section](https://github.com/insolar/observer/blob/master/README.md#build) of this document. Alternatively, you can pull to the existing cloned repo and then build the Node.
+3. Deploy the Node and Node API as described in the [**Configure and deploy the Node**](https://github.com/insolar/observer/blob/master/README.md#configure-and-deploy-the-node) and [**Configure and deploy the Node API**](https://github.com/insolar/observer/blob/master/README.md#configure-and-deploy-the-node-api) sections respectively.
+Check for [new or updated configuration parameters](https://github.com/insolar/observer/wiki/Configuration-parameters) to make sure your current configurations are fully compatible with the updated Node binary files.  
+
+After the update has been completed, run your Node instance.
```
Comment (sofiayem) [inline README.md:189]: This step is part of the deployment sequence. You can drop this sentence.

**Review by sofiayem — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (raproid) [reply README.md:172]: Version does not indeed. But the Node of current version (current Node version as in "current macOS version doesn't support this or that" or "current software version cannot do this or that") does suspend.

https://www.google.com/search?q=%22current+software+version+doesn%27t+support%22&rlz=1C5CHFA_enRU900RU900&oq=%22current+software+version+doesn%27t+support%22&aqs=chrome..69i57j33l7.8065j0j7&sourceid=chrome&ie=UTF-8 732 results is kinda weak, but this is just an exact phrase I googled. I don't get what confuses you here about the phrasing.

Comment (raproid) [reply README.md:172]: I adopted your option, but please open the link and maybe try some more phrases like the one I googled. It's standard English, even in tech docs.

File: README.md (hunk 3 above)
Comment (raproid) [reply README.md:176]: Accepted

File: README.md (hunk 4 above)
Comment (raproid) [reply README.md:178]: Thanks, I was thinking of "stops".

Comment (raproid) [reply README.md:178]: Adopted fully.

File: README.md (hunk 5 above)
Comment (raproid) [reply README.md:180]: thx, adopted fully.

File: README.md (hunk 6 above)
Comment (raproid) [reply README.md:184]: fixed in both instances

File: README.md (hunk 7 above)
Comment (raproid) [reply README.md:185]: done

File: README.md (hunk 8 above)
Comment (raproid) [reply README.md:187]: You mean "Alternatively, you can pull to the existing cloned repo and then build the Node."? But it's just a comment to the step 2. Help me out, why should it take a whole separate step?

File: README.md (hunk 9 above)
Comment (raproid) [reply README.md:189]: Done.

File: README.md (hunk 2 above)
Comment (raproid) [reply README.md:174]: Each time your Node instance connects to the trusted Heavy Material Node, the instance sends its client type (`client_type`), communication protocol version (`heavy_version`), and smart contract version (`contract_version`) to this Heavy Material Node.

Comment (raproid) [reply README.md:174]: "the node" at the end may be confusing seeing how we call the Observer itself "the Node".

Comment (raproid) [reply README.md:174]: Overused Heavy Material Node throughout the instructions, didn't I? You opinion?

Comment (sofiayem) [reply README.md]: A little, yeah

File: README.md (hunk 8 above)
Comment (sofiayem) [reply README.md]: No, I commented on the 187 line.

**Review by sofiayem — APPROVED** [review]

**Review by shayenblue — APPROVED** [review]

**Review by pavlushev — APPROVED** [review]

**Review by LinnikD — APPROVED** [review]
